solidity/contracts/bridge/Fraud.sol

```diff
@@ -230,19 +192,33 @@ library Frauds {
     ///        according to the protocol rules and already proved in the Bridge
     ///      - before a defeat attempt is made the transaction that spends the
     ///        given UTXO must be proven in the Bridge
-    function defeatChallenge(
-        Data storage self,
+    function defeatFraudChallenge(
+        BridgeState.Storage storage self,
         bytes calldata walletPublicKey,
         bytes calldata preimage,
+        bool witness,
         address treasury
```

Comment:
I think we don't need to pass the `treasury` parameter here, as we take it from `BridgeState` structure and we have access to this structure here.
So we can access `treasury` with `self.treasury` here.

Reply:
Good idea! Dropped in 3e3e03be7be628b097d6ec5272098b265ed46805.